apps/web/src/pages/templates/workflow/workflow/node-types/ChannelNode.tsx

```diff
@@ -1,125 +1,82 @@
-import { ChannelTypeEnum, DelayTypeEnum, StepTypeEnum } from '@novu/shared';
 
 import { WorkflowNode } from './WorkflowNode';
 import { useParams } from 'react-router-dom';
-import { IFormStep, ITemplates } from '../../../components/formTypes';
-import { WillBeSentHeader } from '../../digest/WillBeSentHeader';
-
-interface NodeData {
-  Icon: React.FC<any>;
-  label: string;
-  tabKey: ChannelTypeEnum;
-  index: number;
-  testId: string;
-  onDelete: (uuid: string) => void;
-  onAddVariant: (uuid: string) => void;
-  error: string;
-  active?: boolean;
-  channelType: StepTypeEnum;
-  uuid: string;
-  name?: string;
-  content?: ITemplates['content'];
-  htmlContent?: ITemplates['htmlContent'];
-  delayMetadata?: IFormStep['delayMetadata'];
-  digestMetadata?: IFormStep['digestMetadata'];
-}
-
-export default memo(
-  ({ data, selected, id, dragging }: { data: NodeData; selected: boolean; id: string; dragging: boolean }) => {
-    const { getNode, getEdges, getNodes } = useReactFlow();
-    const nodes = useNodes<NodeData>();
-    const thisNode = getNode(id);
-    const isParent = thisNode ? getOutgoers(thisNode, getNodes(), getEdges()).length : false;
-    const noChildStyle = isParent ? {} : { border: 'none', background: 'transparent' };
-    const [count, setCount] = useState(0);
-    const { stepUuid = '' } = useParams<{ stepUuid: string }>();
-
-    useEffect(() => {
-      const items = nodes
-        .filter((node) => node.type === 'channelNode')
-        .filter((node) => {
-          return node.data.channelType === data.channelType;
-        });
-
-      if (items.length <= 1) {
-        setCount(0);
-
-        return;
-      }
-
-      const foundIndex = items.findIndex((node) => node.id === id);
-
-      if (foundIndex === -1) {
-        setCount(0);
-
-        return;
-      }
-
-      setCount(foundIndex + 1);
-    }, [nodes, data, id]);
-
-    const subtitle = useMemo(() => {
-      const content = data.content;
-      if (StepTypeEnum.DELAY === data.channelType) {
-        return delaySubtitle(data);
-      }
-      if (StepTypeEnum.DIGEST === data.channelType) {
-        return <WillBeSentHeader index={data.index} isHighlight={false} />;
-      }
-
-      if (typeof content === 'string') {
-        return content;
-      }
-
-      if (data.channelType === StepTypeEnum.EMAIL) {
-        if (content && content?.length > 0) {
-          return content?.find((item) => item.type === 'text')?.content;
-        }
-        if (data.htmlContent) {
-          return data.htmlContent;
-        }
-      }
+import { INode } from '../../../../../components/workflow/types';
+import { useStepSubtitle } from '../../../hooks/useStepSubtitle';
+
+export default memo((node: INode) => {
+  const { data, id, dragging } = node;
+  const { testId, error, channelType, step, label, index, tabKey, Icon, onAddVariant, onDelete, onEdit } = data;
+  const { active, uuid, name, template } = step;
+  const variantsCount = step.variants?.length;
```

Comment:
before we were passing the `active, uuid, name` fields etc one by one, now we do pass the whole `step` instead